### DISCUSSION TITLE
How is the `defaultValue` prop different from `value`?

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [suren-atoyan/monaco-react#437] [Q&A]
If I set `value` directly, it seems to have the same effect as (and override, if both are present) whatever I pass as `defaultValue`.

How are these different?

**Comment by jgthms** [comment]
AFAIK, `defaultValue` is only read on initialization. This means the editor won't update if you somehow update the value of `defaultValue`.

For example, if you use `const [myCode, setCode] = useState("")` to store the code you want to display, `defaultValue={myCode}` will always stay at `""` even if you update it later.

With `value={myCode}`, it will be updated.
